NameLayerMain/src/vg/civcraft/mc/namelayer/gui/MainGroupGUI.java

```diff
@@ -0,0 +1,1272 @@
+							g.removeMember(p.getUniqueId());
+							p.sendMessage(ChatColor.GREEN + "You have left "
+									+ g.getName());
+							checkRecacheGroup();
+						}
+					}, 11);
+					confirmInv.setSlot(new Clickable(no) {
+
+						@Override
+						public void clicked(Player p) {
+							showScreen();
+						}
+					}, 15);
+					confirmInv.setSlot(new DecorationStack(info), 4);
+					confirmInv.showInventory(p);
+				}
+			};
+		}
+		return c;
+	}
+
+	private Clickable getInfoStack() {
+		Clickable c;
+		ItemStack is = new ItemStack(Material.PAPER);
+		ISUtils.setName(is, ChatColor.GOLD + "Stats for " + g.getName());
+		ISUtils.addLore(is,
+				ChatColor.DARK_AQUA + "Your current rank: " + ChatColor.YELLOW
+						+ getDirectRankName(g.getPlayerType(p.getUniqueId())));
+		boolean hasGroupStatsPerm = gm.hasAccess(g, p.getUniqueId(),
+				PermissionType.getPermission("GROUPSTATS"));
+		if (gm.hasAccess(g, p.getUniqueId(),
+				PermissionType.getPermission("MEMBERS"))
+				|| hasGroupStatsPerm) {
+			ISUtils.addLore(
+					is,
+					ChatColor.AQUA
+							+ String.valueOf(g
+									.getAllMembers(PlayerType.MEMBERS).size())
+							+ " members");
+		}
+		if (gm.hasAccess(g, p.getUniqueId(),
+				PermissionType.getPermission("MODS"))
+				|| hasGroupStatsPerm) {
+			ISUtils.addLore(
+					is,
+					ChatColor.AQUA
+							+ String.valueOf(g.getAllMembers(PlayerType.MODS)
+									.size()) + " mods");
+		}
+		if (gm.hasAccess(g, p.getUniqueId(),
+				PermissionType.getPermission("ADMINS"))
+				|| hasGroupStatsPerm) {
+			ISUtils.addLore(
+					is,
+					ChatColor.AQUA
+							+ String.valueOf(g.getAllMembers(PlayerType.ADMINS)
+									.size()) + " admins");
+		}
+		if (gm.hasAccess(g, p.getUniqueId(),
+				PermissionType.getPermission("OWNER"))
+				|| hasGroupStatsPerm) {
+			ISUtils.addLore(
+					is,
+					ChatColor.AQUA
+							+ String.valueOf(g.getAllMembers(PlayerType.OWNER)
+									.size()) + " owner");
+		}
+		if (hasGroupStatsPerm) {
+			ISUtils.addLore(
+					is,
+					ChatColor.DARK_AQUA
+							+ String.valueOf(g.getAllMembers().size())
+							+ " total group members");
+			ISUtils.addLore(is, ChatColor.DARK_AQUA + "Group owner: "
+					+ ChatColor.YELLOW + NameAPI.getCurrentName(g.getOwner()));
+		}
+		c = new DecorationStack(is);
+		return c;
+	}
+
+	/**
+	 * Utility to get a "nice" version of the rank the given player has in the
+	 * group this gui is based on
+	 * 
+	 * @param uuid
+	 *            Player whose rank name should be checked
+	 * @return Rankname of the given player ready to be put into the gui or null
+	 *         if the given UUID was null or the player didn't have an explicit
+	 *         rank in the group
+	 */
+	private String getRankName(UUID uuid) {
+		if (uuid == null) {
+			return null;
+		}
+		PlayerType pType = g.getPlayerType(uuid);
+		if (pType == null) {
+			return null;
+		}
+		return getDirectRankName(pType);
+	}
+
+	public static String getDirectRankName(PlayerType pType) {
```

Comment:
Hmm, that's used to get a user friendly representation of rank names, so:

MEMBERS --> Member
MODS --> Mod
ADMINS --> Admin
OWNER --> Owner

This used to also do some different stuff when I initially made it, but I later on gutted it. I guess the way it is now, a switch would be the appropriate solution